### ISSUE TITLE
[Meta] Shipper 8.5 - Experimental integration with Filebeat and Metricbeat

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent-shipper#15]
The 8.5 release milestone minimum feature set necessary to use the shipper with metricbeat and filebeat: only support the memory queue and the Elasticsearch output, leaving processors in the beats. The shipper can be enabled by setting an environment variable when starting the agent. A feature flag will be added to the agent policy and Fleet UI in a future iteration.

![image](https://user-images.githubusercontent.com/3466215/159567201-a4105a48-cc49-4e6b-bb55-c289d505b4fd.png)

## Dependencies
The primary dependency of the phase 1 data shipper is the agent control plane. The control plane work  to support the phase 1 shipper is tracked in:

- https://github.com/elastic/elastic-agent/issues/836
  - https://github.com/elastic/elastic-agent/issues/214

## Implementation
The implementation tasks for the shipper are broken into several independent feature groups, listed below. Tasks will be converted to issues as the implementation progresses.

- [x] https://github.com/elastic/elastic-agent-shipper/issues/4
- [x] https://github.com/elastic/elastic-agent-shipper/issues/5
- [x] https://github.com/elastic/elastic-agent-shipper/issues/6
- [x] https://github.com/elastic/elastic-agent-shipper/issues/13
  - [x] https://github.com/elastic/elastic-agent-shipper/issues/46
  - [x] https://github.com/elastic/elastic-agent-shipper/issues/101
  - [x] Test the shipper's V2 agent control client once the server is complete.
- [x] https://github.com/elastic/elastic-agent-shipper/issues/7
- [x] https://github.com/elastic/elastic-agent-shipper/issues/83
- [x] https://github.com/elastic/elastic-agent-shipper/issues/8
  - [x] elastic/elastic-agent-shipper#22
  - [x] elastic/elastic-agent-shipper#23
  - [x] elastic/elastic-agent-shipper#24
  - [x] elastic/elastic-agent-shipper#34
    - [x] https://github.com/elastic/elastic-agent-shipper/issues/84
    - [x] The API specification is updated with any changes or improvements that were necessary.
    - [x] https://github.com/elastic/elastic-agent-shipper/issues/62
    - [x] A reference API client implementation exists and integrates with the server side implementation. 
    - [x] https://github.com/elastic/elastic-agent-shipper/issues/64
    - [x] https://github.com/elastic/beats/issues/32081
    - [x] https://github.com/elastic/elastic-agent-shipper/issues/79
    - [x] https://github.com/elastic/beats/issues/32814
    - [x] An integration test suite exists to verify that the client and server implementations interact correctly, with emphasis on error conditions: partially accepted batches, full queues, server and client side restarts, etc.
      - [x] https://github.com/elastic/beats/issues/32329
      - [x] https://github.com/elastic/elastic-agent-shipper/issues/98
- [x] https://github.com/elastic/elastic-agent-shipper/issues/9
  - [x] https://github.com/elastic/beats/pull/32541
  - [x] https://github.com/elastic/beats/issues/32329
  - [x] https://github.com/elastic/elastic-agent-shipper/issues/98

## QA / Testing Needs
Deferred to 8.6

## Documentation
N/A - The shipper will be experimental and enabled via an environment variable when starting the agent.

**Comment by nimarezainia** [comment]
@cmacknz @pierrehilbert is this meta issue what we should track for experimental availability of shipper in 8.5?

**Comment by cmacknz** [comment]
Yes this is the issue for the 8.5 milestone.

**Comment by cmacknz** [comment]
Only outstanding issues to close this is the Elasticsearch output https://github.com/elastic/elastic-agent-shipper/issues/10 and standalone mode which is already in review https://github.com/elastic/elastic-agent-shipper/issues/83

**Comment by cmacknz** [comment]
Closing, all outstanding work moved to the next milestone: https://github.com/elastic/elastic-agent-shipper/issues/16
